temporal/server_test.go

```diff
@@ -37,16 +43,89 @@ import (
 	"go.temporal.io/server/tests/testutils"
 )
 
-// TestNewServer verifies that NewServer doesn't cause any fx errors
+// TestNewServer verifies that NewServer doesn't cause any fx errors, and that there are no unexpected error logs after
+// running for a few seconds.
 func TestNewServer(t *testing.T) {
+	t.Parallel()
+
+	cfg := loadConfig(t)
+	logger := newErrorLogDetector(t)
+
+	server, err := NewServer(
+		WithConfig(cfg),
+		WithLogger(logger),
+	)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		assert.NoError(t, server.Stop())
+	})
+	require.NoError(t, server.Start())
+	time.Sleep(10 * time.Second)
+}
+
+func loadConfig(t *testing.T) *config.Config {
+	cfg := loadSQLiteConfig(t)
+	setTestPorts(cfg)
+	return cfg
+}
+
+// loadSQLiteConfig loads the config for the sqlite persistence store. We use sqlite because it doesn't require any
+// external dependencies, so it's easy to run this test in isolation.
+func loadSQLiteConfig(t *testing.T) *config.Config {
 	configDir := path.Join(testutils.GetRepoRootDirectory(), "config")
 	cfg, err := config.LoadConfig("development-sqlite", configDir, "")
 	require.NoError(t, err)
+
 	cfg.DynamicConfigClient.Filepath = path.Join(configDir, "dynamicconfig", "development-sql.yaml")
-	_, err = NewServer(
-		ForServices(DefaultServices),
-		WithConfig(cfg),
-	)
-	assert.NoError(t, err)
-	// TODO: add tests for Server.Run(), etc.
+	return cfg
+}
+
+// setTestPorts sets the ports of all services to something different from the default ports, so that we can run the
+// tests in parallel.
+func setTestPorts(cfg *config.Config) {
+	port := 10000
```

Comment:
doesn't this need a new port every time? maybe we need an global atomic int?

Reply:
It's ok for this to be hard-coded until https://github.com/temporalio/temporal/issues/4329 is done. We'll only conflict if the host has something else running in the low 10000 range because this logic isn't reused for any other tests. Maybe I'm misunderstanding the question, though. What's the issue you're seeing?

Reply:
no issue yet since there's only one parallel test in this package, just that the stated goal is to allow running tests in parallel, but it can't do that if it always picks the same ports

Reply:
The goal is to allow it to run in parallel with any new tests added in this package that don't also start a server, but yeah it does nothing right now since there's only one test in this package.

Reply:
that don't also start a server in the same range*

Reply:
Like basically I'm marking it with `t.Parallel` in case we add another test to the `temporal` package later for something like `temporal.InterruptCh`